ui/app/utils/classes/rolling-array.js

```diff
@@ -0,0 +1,45 @@
+// An array with a max length.
+//
+// When max length is surpassed, items are removed from
+// the front of the array.
+
+// Using Classes to extend Array is unsupported in Babel so this less
+// ideal approach is taken: https://babeljs.io/docs/en/caveats#classes
+export default function RollingArray(maxLength, ...items) {
+  const array = new Array(...items);
+  array.maxLength = maxLength;
+
+  // Capture the originals of each array method, but
+  // associate them with the array to prevent closures.
+  array._push = array.push;
+  array._splice = array.splice;
+  array._unshift = array.unshift;
+
+  array.push = function(...items) {
+    const returnValue = this._push(...items);
+
+    const surplus = this.length - this.maxLength;
+    if (surplus > 0) {
+      this.splice(0, surplus);
+    }
+
+    return Math.min(returnValue, this.maxLength);
+  };
+
+  array.splice = function(...args) {
+    const returnValue = this._splice(...args);
+
+    const surplus = this.length - this.maxLength;
+    if (surplus > 0) {
+      this._splice(0, surplus);
+    }
+
+    return returnValue;
+  };
+
+  array.unshift = function() {
```

Comment:
Probably not a use case for it, but could be cool to support push and unshift (and probably not splice? or just push out both sides with splice?) and then you remove the surplus from the opposite side so that it's a SlideArray.

Reply:
Hey, sorry bit late here, if you are avoiding closures, would `Array.prototype.push.apply(this, arguments)` help here? It would mean you wouldn't have to add your `_push` etc (polluting the array)

Reply:
Normally I hesitate to do that because there is no guarantee that `array.push === Array.prototype.push`, but in this case, the Array constructor is being invoked directly, so you're totally right that I can just use the methods on the prototype.

Good call, I'll make this change in my current branch 👍